tests/code/test_eventList.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+from app.logic.events import groupEventsByCategory, groupEventsByProgram
+from peewee import DoesNotExist
+from app.models.program import Program
+from app.models.event import Event
+from app.models.programEvent import ProgramEvent
+from app.models.term import Term
+
+@pytest.mark.integration
+def test_termDoesNotExist():
+    with pytest.raises(DoesNotExist):
+        groupedEvents = groupEventsByCategory(7)
+        groupedEvents2 = groupEventsByCategory("khatts")
+
+@pytest.mark.integration
+def test_groupEventsByProgram():
+    studentLedEvents = (Event.select(Event, Program.id.alias("program_id"))
+                             .join(ProgramEvent)
+                             .join(Program)
+                             .where(Program.isStudentLed,
+                                    Event.term == 1))
+    groupedEvents = groupEventsByProgram(studentLedEvents)
+    assert groupedEvents == {Program.get_by_id(1): [Event.get_by_id(1), Event.get_by_id(2)] , Program.get_by_id(2): [Event.get_by_id(4)]}
+
+@pytest.mark.integration
+def test_groupEventsByCategory():

```

Comment:
This function has a lot of other queries that are a little different. You should set up test data so that this test checks them all, not just studentLedEvents. And actually, it's not really testing that one either, is it, just making sure there aren't any errors in execution.

Reply:
Made tests for all queries